frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -0,0 +1,244 @@
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'IncomeDetails',
+  mixins: [validationMixin],
+  components: {
+    ModalBaseTemplate,
+    InputAmount,
+    TextWho,
+    Tooltip,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  data () {
+    return {
+      ephemeral: {
+        formFeedbackMsg: {}
+      },
+      form: {
+        incomeDetailsType: null,
+        amount: null
+      }
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'groupMembers',
+      'groupSettings',
+      'memberProfile',
+      'groupMincomeFormatted',
+      'groupMincomeSymbolWithCode',
+      'ourUsername',
+      'groupMembersByUsername'
+    ]),
+    memberGroupProfile () {
+      const profile = this.memberProfile(this.ourUsername) || {}
+      return profile.groupProfile || {}
+    },
+    needsIncome () {
+      return this.form.incomeDetailsType === 'incomeAmount'
+    },
+    whoIsPledging () {
+      return this.groupMembersByUsername.filter(username => this.groupMembers[username].groupProfile.incomeDetailsType === 'pledgeAmount' && username !== this.ourUsername)
+    }
+  },
+  created () {
+    const incomeDetailsType = this.memberGroupProfile.incomeDetailsType
+    if (incomeDetailsType) {
+      this.form.incomeDetailsType = incomeDetailsType
+      this.form.amount = this.memberGroupProfile[incomeDetailsType]
+    }
+  },
+  methods: {
+    resetAmount () {
+      this.form.amount = this.form.incomeDetailsType === this.memberGroupProfile.incomeDetailsType ? this.memberGroupProfile[this.memberGroupProfile.incomeDetailsType] : ''
+      this.$v.form.$reset()
+    },
+    closeModal () {
+      this.$refs.modal.close()
+    },
+    async submit () {
+      if (this.$v.form.$invalid) {
+        alert(L('Invalid payment info'))
+        return
+      }
+      try {
+        const incomeDetailsType = this.form.incomeDetailsType
+        const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
+          {
+            incomeDetailsType,
+            [incomeDetailsType]: +this.form.amount
+          },
+          this.$store.state.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', groupProfileUpdate)
+        this.closeModal()
+      } catch (e) {
+        console.error('setPaymentInfo', e)
+        this.ephemeral.formFeedbackMsg = {
+          text: L(`Failed to update income details. Error: ${e.message}`),
+          class: 'has-text-danger'
+        }
+      }
+    }
+  },
+  validations: {
+    form: {
+      incomeDetailsType: {
+        [L('This field is required')]: required
+      },
+      amount: {
+        [L('This field is required')]: required,
+        [L("It seems your income is not lower than the group's mincome.")]: function (value) {
+          if (this.needsIncome) {
+            return value < this.groupSettings.mincomeAmount - 1
+          }
+          return true
+        }
```

Comment:
Needs the other validation from Contributions.vue, to make sure no negative numbers